wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -371,7 +371,7 @@ data WalletAddress = WalletAddress
   , addrBalance       :: !(V1 Core.Coin)
   , addrUsed          :: !Bool
   , addrChangeAddress :: !Bool
-  } deriving (Show, Generic)
+  } deriving (Show, Eq, Generic)
```

Comment:
I am not sure does it makes sense to add `Eq` here ? Some might think that `WalletAddress` should implement `Eq` in a way where `walletId`s are the same. It might be confusing as the name suggests that it is an address, and two addresses should be the same depending on their IDs, excluding other metadata.

But I might be wrong.

Reply:
I'm not sure I understand. The datatype will be equal if all the fields inside are equal. Not really sure why that's not valid.

Reply:
I think that the `WalletAddress` type represents both the address + metadata about it. If you wanted to compare two `WalletAddresses` and only focus on a field, there's always ```(==) `on` addrId```